### PR TITLE
UsageStats: track enabled features

### PR DESCRIPTION
**Pull request by ArturWierzbicki**
Part 1 of https://github.com/grafana/grafana/issues/47127

This PR adds enabled features to the usage stats report
image

### CODE REVIEW

File: pkg/services/featuremgmt/service.go [hunk 1]
```diff
@@ -74,3 +75,12 @@ func ProvideManagerService(cfg *setting.Cfg, licensing models.Licensing) (*Featu
 func ProvideToggles(mgmt *FeatureManager) FeatureToggles {
 	return mgmt
 }
+
+func (fm *FeatureManager) GetUsageStats(ctx context.Context) map[string]int {
+	enabled := fm.GetEnabled(ctx)
+	stats := make(map[string]int, len(enabled))
+	for featureName := range enabled {
+		stats[fmt.Sprintf("stats.features.%s.count", featureName)] = 1
```
Comment (ArturWierzbicki) [inline pkg/services/featuremgmt/service.go:83]: We don't have a strict naming convention for feature names, e.g.:

- `database_metrics`
- `service-accounts`
- `envelopeEncryption`

I wonder if I should convert them all to snake case or would simply lower casing be enough.

Comment (ArturWierzbicki) [reply pkg/services/featuremgmt/service.go:83]: Also, some names are pretty long: `azureMonitorResourcePickerForMetrics`, need to figure out the max length for metric name

**Comment by FZambia** [comment]
Artur, did you mean this should be backported to v8.4.x?

**Comment by ArturWierzbicki** [comment]
@FZambia I believe that PRs targeting 8.5.x at this point in time need a 8.5.x backport label. This functionality will be useful in 8.4 as well though so might as well target a lower version

**Review by ying-jeanne — COMMENTED** [review]
the code LGTM, out of curiosity, the feature that activated is not already stored in database of HG API server? why we need to gather them from metrics? @sakjur since you are also working on refactor of the usage stats now :) [PR](https://github.com/grafana/grafana/pull/47381)

**Review by ying-jeanne — APPROVED** [review]

File: pkg/infra/usagestats/service/usage_stats.go [hunk 2]
```diff
@@ -230,6 +230,11 @@ func (uss *UsageStats) GetUsageReport(ctx context.Context) (usagestats.Report, e
 
 	metrics["stats.uptime"] = int64(time.Since(uss.startTime).Seconds())
 
+	featureUsageStats := uss.features.GetUsageStats(ctx)
```
Comment (sakjur) [inline pkg/infra/usagestats/service/usage_stats.go:233]: Would it be possible to use https://github.com/grafana/grafana/blob/725dbf8d95cdee03bdee12f168f9c84b185c1d6d/pkg/infra/usagestats/service.go#L18 https://github.com/grafana/grafana/blob/725dbf8d95cdee03bdee12f168f9c84b185c1d6d/pkg/infra/usagestats/service.go#L24 instead?

**Comment by sakjur** [comment]
> the feature that activated is not already stored in database of HG API server? why we need to gather them from metrics?

Artur probably has a better response to this than I do, my take on it is that these metrics are available for the on prem installations that share the usage stats with us as well, and they reflect the reality and not the expected state (unlike the database, which can lie to you if say the application hasn't been restarted or there's some semi-hidden typo).

Useful to have both.

> @sakjur since you are also working on refractory of the usage stats now :)

Thank you, this is not a problem for me to merge with even if you chose not to go with my suggestion 👆 (if you do, they won't conflict or break each other), @ArturWierzbicki you prob want to merge before me in that case.

File: pkg/infra/usagestats/service/usage_stats.go (hunk 2 above)
Comment (sakjur) [reply pkg/infra/usagestats/service/usage_stats.go:233]: Ah, yeah, let's merge this and I'll see if my refactor makes that possible then 🙂

**Comment by ArturWierzbicki** [comment]
Thanks for reviews @ying-jeanne @sakjur !

**Pull request merged**